### ISSUE TITLE
Investigate Firefox sound issues

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#815]
See if jrx's sound issue can be replicated, then fix it if it's easy or make a separate issue if it's hard.

> jrx: dh - i am not get the sound warnings for time to act on chrome, back to firefox
jrx: backwards, not getting the sound warnings on firefox, back to chrome
jrx: sound working fine on other firefox windows

**Comment by ctm** [comment]
Sound is a tricky issue. Similar to pop-ups, browsers can choose to disallow sound. On top of that, although some browsers will do something to show you that sound is being blocked, I believe it's possible to (perhaps inadvertently) have a configuration where they'll block and not notify you.

Luckily, apps that are trying to play sound can detect (at least under most circumstances) when sound is being blocked. However, mb2 doesn't currently do that because I didn't know enough about web programming in general and web programming with Rust in specific to do so when I added sound.

A year and a half ago I did some exploration and created the "Add Safari Sound workaround" issue (#394), but I never did anything with it. I now know enough to be able to detect when sound isn't playing and bring up a dialog to tell the user. I suspect I can even make it so that clicking the OK button will allow sound for the remainder of the time that window is open, but that's based on "book knowledge" rather than actual experience.

So, I'll continue working on this, but only while everything still seems easy. If it gets hard, I'll get to a good stopping point and if that stopping point is an improvement, I'll release what I've done. I'll make a new issue for the hard stuff.
